src/synapse/azext_synapse/custom.py

```diff
@@ -121,59 +152,123 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, account_u
+    return client.check_name_availability(name, "Microsoft.Synapse/workspaces")
+
+
 def get_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name):
     return client.get(resource_group_name, workspace_name, spark_pool_name)
 
 
-def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, location,
-                      spark_version, node_size=NodeSize.medium.value, node_count=3,
-                      node_size_family=NodeSizeFamily.memory_optimized.value, auto_scale_enabled=True,
-                      min_node_count=3,
-                      max_node_count=40, auto_pause_enabled=True, delay_in_minutes=15, spark_events_folder="/events",
-                      library_requirements_filename=None, library_requirements_content=None,
-                      default_spark_log_folder="/logs", force=False, tags=None, no_wait=False):
+def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name,
+                      spark_version, node_size, node_count,
+                      node_size_family=NodeSizeFamily.memory_optimized.value, enable_auto_scale=None,
+                      min_node_count=None, max_node_count=None,
+                      enable_auto_pause=None, delay_in_minutes=None, spark_events_folder="/events",
+                      library_requirements_file=None,
+                      default_spark_log_folder="/logs", tags=None, no_wait=False):
+    # get the location of the workspace
+
+    workspace_client = cf_synapse_client_workspace_factory(cmd.cli_ctx)
+    workspace_object = workspace_client.get(resource_group_name, workspace_name)
+    location = workspace_object.location
+
     big_data_pool_info = BigDataPoolResourceInfo(location=location, spark_version=spark_version, node_size=node_size,
                                                  node_count=node_count, node_size_family=node_size_family,
                                                  spark_events_folder=spark_events_folder,
                                                  default_spark_log_folder=default_spark_log_folder, tags=tags)
-    if auto_scale_enabled:
-        big_data_pool_info.auto_scale = AutoScaleProperties(enabled=auto_scale_enabled, min_node_count=min_node_count,
-                                                            max_node_count=max_node_count)
-    if auto_pause_enabled:
-        big_data_pool_info.auto_pause = AutoPauseProperties(enabled=auto_pause_enabled,
-                                                            delay_in_minutes=delay_in_minutes)
-
-    if library_requirements_filename or library_requirements_content:
-        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_filename,
+
+    big_data_pool_info.auto_scale = AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
+                                                        max_node_count=max_node_count)

```

Comment:
I would write below to avoid creating `AutoScaleProperties` even it is not set. Just throw my thought. It is okay to keep the current logic.

```
big_data_pool_info.auto_scale = enable_auto_scale and  AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
                                                        max_node_count=max_node_count)
```

or 

```
big_data_pool_info.auto_scale = enable_auto_scale and min_node_count and max_node_count and  AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
                                                        max_node_count=max_node_count)
```

or 

```
if not all_or_none(enable_auto_scale, min_node_count, max_node_count):
    raise some error
big_data_pool_info.auto_scale =  AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
                                                        max_node_count=max_node_count)
```

Reply:
In fact I want to create AutoScaleProperties even it is not set. Otherwise the there is something wrong in portal. And I have checked in portal, it will create AutoScaleProperties even it is not set. We can talk more details offline.

---
In reply to: [418058271](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r418058271) [](ancestors = 418058271)

Reply:
As disscussed offline let's keep it.

---
In reply to: [421222862](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r421222862) [](ancestors = 421222862,418058271)